### DISCUSSION TITLE
A pattern for appending a default/base value when using `next/head`

### DISCUSSION THREAD

**Discussion by BigglesZX** [vercel/next.js#21580] [Help] [answered]
I'm just getting started with Next.js but really enjoying the experience so far – so, thanks!

I'm using `next/head` to set a default page title in my custom app, and then override that title in my pages. Simple enough.

However, I was wondering if there's a sensible or recommended way to **append** my default/base title to the titles set by my pages, **without** having to share that default title data with every page. In Django for example, the world I'm coming from, I could call `block.super` inside my `title` block in a child template, which would bring in the value from the parent template, allowing prepending/appending etc.

So the notional desired outcome might be:

**Default title set in `_app.js`:** `My Site`
**Title set by "About" page:** `About`
**Desired resulting value of `<title>` on "About" page:** `About | My Site`

I know I could achieve this if I shared that default title string with my page, e.g. in a React `context`, but I'm interested to hear if there's a way to avoid that, since I want to do this on almost every page. If there's a way to do it without each page needing access to the default title data, that would be great.

Any advice much appreciated. Thanks for reading!

**Comment by michrome** [comment, marked as answer]
Hey @BigglesZX - I'd approach this with a layout component e.g.

```javascript
// components/layout.js
import Head from "next/head";

export default function Layout({ title, children }) {
  const fullTitle = [title, "My Site"].join(" | ");
  return (
    <>
      <Head>
        <title>{fullTitle}</title>
      </Head>
      {children}
    </>
  );
}
```

```javascript
// pages/my-page.js
import Layout from "../components/layout";

export const title = "My Page";

export default function MyPage() {
  return <Layout title={title}>Content here</Layout>;
}
```

**Reply by BigglesZX** [reply]
Thanks @michrome – this has got me on the right track. If I didn't want to wrap my page components I could presumably also use a custom `Title` component which used `Head` to set the page title as in your example. That way only the `Title` component would need to be aware of the default data, just as in your example.

**Comment by BigglesZX** [comment]
For future readers, this is what I ended up with thanks to prompting by @michrome:

```javascript
// pages/_app.js (vastly simplified)
import Head from 'next/head';
import Title from 'components/Title';

const MyApp = ({ Component, pageProps }) => {
    return (
        <>
            <Head>
                <meta name="description" content={config.metaDescription} />
            </Head>
            <Title />
            <Component {...pageProps}></Component>
        </>
    );
};

export default MyApp;
```

```javascript
// components/Title.js
import Head from 'next/head';

const Title = ({ children }) => {
    const defaultTitle = 'My Site';
    return (
        <Head>
            <title>{children ? `${children} › ${defaultTitle}` : defaultTitle}</title>
        </Head>
    );
};

export default Title;
```

```javascript
// pages/colophon.js
import Title from 'components/Title';

const Colophon = () => {
    return (
        <>
            <Title>Colophon</Title>
            <article>
                <h1>Colophon</h1>
                Lorem ipsum dolor etc...
            </article>
        </>
    );
};

export default Colophon;
```
